ci/docker/linux-apt-python-3.dockerfile

```diff
@@ -39,6 +39,12 @@ RUN pip install \
     -r arrow/python/requirements-build.txt \
     -r arrow/python/requirements-test.txt
 
+ARG numba
+COPY ci/scripts/install_numba.sh /arrow/ci/scripts/
+RUN if [ "${numba}" != "" ]; then \
+        /arrow/ci/scripts/install_numba.sh ${numba} \
+    ; fi
```

Comment:
Not sure this is ideal, but It wasn't able to create a child Dockerfile with this file as BASE (I may be misunderstanding how these images are managed).

Reply:
A good example on child images/services is the `ubuntu-r` one, which builds on the `ubuntu-cpp` dockerfile in it's own `linux-apt-r.dockerfile` 
https://github.com/apache/arrow/blob/master/docker-compose.yml#L1234-L1242

Reply:
Hmm... can you suggest a patch? I already tried to follow this pattern and didn't succeed. I got this error:
```
DEBUG:archery:Executing `['docker-compose', '--file', '/home/antoine/arrow/dev/docker-compose.yml', 'build', '--build-arg', 'BUILDKIT_INLINE_CACHE=1', 'ubuntu-cuda-python']`
Building ubuntu-cuda-python
[+] Building 0.4s (3/3) FINISHED                                                                                                                                     
 => [internal] load build definition from linux-apt-python-numba.dockerfile                                                                                     0.0s
 => => transferring dockerfile: 1.04kB                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                               0.0s
 => => transferring context: 35B                                                                                                                                0.0s
 => ERROR [internal] load metadata for docker.io/apache/arrow-dev:amd64-linux-apt-python-3                                                                      0.4s
------
 > [internal] load metadata for docker.io/apache/arrow-dev:amd64-linux-apt-python-3:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/apache/arrow-dev:amd64-linux-apt-python-3: not found
ERROR: Service 'ubuntu-cuda-python' failed to build : Build failed
Error: `docker-compose --file /home/antoine/arrow/dev/docker-compose.yml build --build-arg BUILDKIT_INLINE_CACHE=1 ubuntu-cuda-python` exited with a non-zero exit code 1, see the process log above.

```

Reply:
(side note: it's not help ful that browsing to `docker.io/apache/arrow-dev` does not let me view the available images/tags)